compiler-core/src/type_/environment.rs

```diff
@@ -587,4 +636,70 @@ impl<'a> Environment<'a> {
             .map(|t| t.to_string())
             .collect()
     }
+
+    /// Checks that the given patterns are exhaustive for given type.
+    /// Currently only performs exhaustiveness checking for custom types,
+    /// only at the top level (without recursing into constructor arguments).
+    pub fn exhaustive(
+        &mut self,
+        patterns: Vec<Pattern<PatternConstructor, Arc<Type>>>,
+        value_typ: Arc<Type>,
+    ) -> Option<Vec<String>> {
```

Comment:
I think the name doesn't line up quite with the return value here. Either we could call it `unmatched_constructors` and keep the same return, or we could call it `exhaustive` or `check_exhaustiveness` and return `Result<(), Vec<String>>`.

 I think the altter latter would probably be more in keeping with the existing codebase. Makes it clearer it's the sad path.

Reply:
Updated!

Reply:
This was done in https://github.com/gleam-lang/gleam/pull/1462/commits/f931228a71f2f517a836f33b1a55b21a5c6d2938